crates/mdb/src/lib.rs

```diff
@@ -207,7 +207,17 @@ impl SubscriberConfig {
         debug!("Creating {}...", any::type_name::<Self>());
         unsafe {
             let raw_on_message = Box::into_raw(Box::new(on_message));
-            let on_message = Deferred::new(raw_on_message);
+            // TODO: Verify that panic is sound.
```

Comment:
What does happen if this panics? Doesn't the whole program panic nothing afterwards is unsound?

Reply:
The default panic behavior is to unwind, and it is importantly that resources can be cleaned up without UB. Furthermore it is possible to catch an unwind and continue despite it entirely in safe code.

The rustonomicon has a short chapter on exception safety that's worth reading: https://doc.rust-lang.org/nomicon/exception-safety.html

This blog post isn't useful but I remember thinking it was interesting: https://smallcultfollowing.com/babysteps/blog/2024/05/02/unwind-considered-harmful/

Reply:
I remember having read that blogpost a while ago, it is interesting. Personally, I never liked unwinding in any language. 
How would you go about verifying that it's sound to panic? Manually panicing everywhere and try to "catch" induce UB? Can Miri help?

Reply:
I thought some more about this and the only functions that can panic are the ones in the match statement and, unless the C library does something completely crazy, I think the cited reasoning covers it. And to some extent we must be able to trust the C library to behave, or we can never make any safe abstraction. So, I removed the todo.

>             // * This function panics; since the user doesn't get a config back the config either
>             //   is leaked or it wasn't created. In either case the pointer will never be
>             //   dereferenced.